pyiceberg/table/__init__.py

```diff
@@ -1522,8 +1555,9 @@ def plan_files(self) -> Iterable[FileScanTask]:
                     data_entry,
                     positional_delete_entries,
                 ),
+                residual=residual,
```

Comment:
Instead of carrying the residuals all the way through, I think we can compute them at this point instead:
```suggestion
                residual=residual_evaluators[manifest.partition_spec_id](data_entry.data_file.partition),
```
If a whole manifest is rejected, we don't even compute the evaluator itself.

Reply:
@Fokko as manifest isn't accessible there can we instead write -
`residual=residual_evaluators[data_entry.data_file.spec_id](data_entry.data_file.partition),`

Reply:
@tusharchou I don't really think this one through. Without any additional changes, this would look like:

```python
        return [
            FileScanTask(
                data_entry.data_file,
                delete_files=_match_deletes_to_data_file(
                    data_entry,
                    positional_delete_entries,
                ),
                residual=residual_evaluators[data_entry.data_file.spec_id](data_entry.data_file).residual_for(
                    data_entry.data_file.partition
                ),
            )
            for data_entry in data_entries
        ]
```

Where we re-create the evaluator all the time